### PR TITLE
feat: switch html parser

### PR DESCRIPTION
**Pull request by viceice**
## Changes:

Use `node-html-parser` again to parse html, as the memory overhead is ~20% less in jest tests. Partial reverts #7092

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added unit tests, or
- [ ] No new tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/util/html.ts [hunk 1]
```diff
@@ -1,5 +1,9 @@
-import { JSDOM } from 'jsdom';
+// import { JSDOM } from 'jsdom';
+import { HTMLElement, parse as _parse } from 'node-html-parser';
+
+export { HTMLElement };
 
 export function parse(html: string): HTMLElement {
-  return new JSDOM(html).window.document.body;
+  return _parse(html);
+  // return new JSDOM(html).window.document.body;
```
Comment (rarkins) [inline lib/util/html.ts:8]:
```suggestion
```

File: lib/util/html.ts [hunk 2]
```diff
@@ -1,5 +1,9 @@
-import { JSDOM } from 'jsdom';
+// import { JSDOM } from 'jsdom';
```
Comment (rarkins) [inline lib/util/html.ts:1]:
```suggestion
```

**Review by rarkins — CHANGES_REQUESTED** [review]

File: lib/datasource/pypi/index.ts [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import url from 'url';
 import changelogFilenameRegex from 'changelog-filename-regex';
 import { logger } from '../../logger';
-import { parse } from '../../util/html';
+import { HTMLElement, parse } from '../../util/html';
```
Comment (rarkins) [inline lib/datasource/pypi/index.ts:4]: (a) how was this used previously if it wasn't imported?
(b) is it needed explicitly here anyway, if the function declares its return type?

Comment (viceice) [reply lib/datasource/pypi/index.ts:4]: It's because the global `HTMLElement` is incompatible to the on of `node-html-parser`

**Review by rarkins — DISMISSED** [review]

**Comment by viceice** [comment]
Removed explicit types, so `HTMLElement` is no longer needed beside tests

**Comment by viceice** [comment]
`jsdom` is still referenced by `jest`, but it should automatically excluded on production builds.

**Review by rarkins — APPROVED** [review]

**Pull request merged**